public/scripts.js

```diff
@@ -8,36 +8,164 @@
     })
   }
 
-  const getVouchers = function () {
-    return $.get('/vouchers.json')
+  const ProductModel = function (price, countChangeHandler) {
+    this.price = 0
+    this.count = 1
+
+    const $productsCountInput = $('#products-count')
+
+    this.init = function (price) {
+      const _self = this
+      _self.price = price
+      _self.count = $productsCountInput.val()
+      _self.countChangeHandler = countChangeHandler
+
+      $productsCountInput.on('change', function () {
+        _self.count = $(this).val()
+        _self.countChangeHandler(_self.count, _self.price)
+      })
+
+      return _self
+    }
+
+    this.getPrice = function () {
+      return this.price
+    }
+
+    this.getCount = function () {
+      return this.count
+    }
+
+    return this.init(price)
   }
 
-  Voucherify.initialize(clientConfig.clientApplicationId, clientConfig.clientPublicKey)
-  Voucherify.setIdentity('gustav@purpleson.com')
-  Voucherify.render('#voucher-checkout', {
-    textPlaceholder: 'e.g. Testing7fjWdr',
-    onValidated: function (res) {
-      const voucherCode = $('#voucher-checkout .voucherifyCode').val()
-      const trackingId = res.tracking_id
-
-      if (res.valid && confirm('Czy chcesz wykorzystać ten Voucher?')) {
-        redeemVoucher(voucherCode, trackingId)
-          .done(function (res) {
-            console.log(res)
-          })
-          .fail(function (err) {
-            console.log(err)
-          })
+  const VoucherCode = function (discountChangeHandler) {
+    this.voucherCode = null
+
+    this.trackingId = null
+
+    this.valid = false
+
+    this.getCode = function () {
+      return this.voucherCode
+    }
+
+    this.getTrackingId = function () {
+      return this.trackingId
+    }
+
+    this.isValid = function () {
+      return this.valid === true
+    }
+
+    this.onValidatedHandler = function (res) {
+      this.voucherCode = $('#voucher-checkout .voucherifyCode').val()
+      this.trackingId = res.tracking_id
+      this.valid = res.valid
```

Comment:
You can assume it's always true, because `onValidated` is only called when result is valid. Check out this code: https://github.com/rspective/voucherify.js/blob/master/lib/voucherify.js#L346